src/coredump/coredump.c

```diff
@@ -1023,84 +1042,86 @@ static int send_iovec(const struct iovec_wrapper *iovw, int input_fd) {
         return 0;
 }
 
-static int gather_pid_metadata(char *context[_CONTEXT_MAX], struct iovec_wrapper *iovw) {
+static int gather_pid_metadata_from_argv(struct iovec_wrapper *iovw, ProcessInfoCache *cache,
+                                         int argc, char **argv) {
 
-        /* Note that if we fail on oom later on, we do not roll-back changes to the iovec
-         * structure. (It remains valid, with the first n_iovec fields initialized.) */
+        /* We gather all metadata that were passed via argv[] into an array of iovecs that
+         * we'll forward to the socket unit */
 
-        uid_t owner_uid;
-        pid_t pid;
+        int i, r, signo;
         char *t;
-        const char *p;
-        int r, signo;
 
-        r = parse_pid(context[CONTEXT_PID], &pid);
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse PID \"%s\": %m", context[CONTEXT_PID]);
+        if (argc < _META_ARGV_MAX)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Not enough arguments passed by the kernel (%i, expected %i).",
+                                       argc, _META_ARGV_MAX);
 
-        r = get_process_comm(pid, &context[CONTEXT_COMM]);
-        if (r < 0)
-                return log_error_errno(r, "Failed to get COMM: %m");
+        for (i = 0; i < _META_ARGV_MAX; i++) {
 
-        r = get_process_exe(pid, &context[CONTEXT_EXE]);
-        if (r < 0)
-                log_warning_errno(r, "Failed to get EXE, ignoring: %m");
-
-        if (cg_pid_get_unit(pid, &context[CONTEXT_UNIT]) >= 0) {
-                if (!is_journald_crash((const char**) context)) {
-                        /* OK, now we know it's not the journal, hence we can make use of it now. */
-                        log_set_target(LOG_TARGET_JOURNAL_OR_KMSG);
-                        log_open();
-                }
+                t = argv[i];
 
-                /* If this is PID 1 disable coredump collection, we'll unlikely be able to process it later on. */
-                if (is_pid1_crash((const char**) context)) {
-                        log_notice("Due to PID 1 having crashed coredump collection will now be turned off.");
-                        disable_coredumps();
+                switch (i) {
+                case META_ARGV_TIMESTAMP:
+                        /* The journal fields contain the timestamp padded with six
+                         * zeroes, so that the kernel-supplied 1s granularity timestamps
+                         * becomes 1µs granularity, i.e. the granularity systemd usually
+                         * operates in. */
+                        t = strjoina(argv[i], "000000");
```

Comment:
repeat after me: i shall not do alloca() inside a loop, i shall not do alloca() inside a loop

Reply:
obviously it can be only executed once per loop, but okay.